minion/server/routes.py

```diff
@@ -10,25 +10,29 @@
 import re
 import mongoengine
 from flask import Blueprint, request, jsonify, render_template
+from jinja2.exceptions import UndefinedError
 
 from .models import Job, mongo_to_dict
 from .core import workers, stop_server, ensure_mongo
 import pymongo
 
 api = Blueprint("api", __name__)
 
+@ensure_mongo
 def get_jobs_page(page=1, page_size=10):
     """Returns all jobs from the given page."""
     return Job.objects.paginate(page=page, per_page=page_size)
 
 
 @api.route("/", methods=["GET"])
-@ensure_mongo
 def index():
     """Serve index HTML."""
     page = int(request.args.get("page", 1))
-    return render_template("index.html", jobs=get_jobs_page(page=page))
-
+    
+    try:
+        return render_template("index.html", jobs=get_jobs_page(page=page))
+    except UndefinedError:
```

Comment:
Well, that's not exactly what I meant and what's sane to do here.

What'd be much better is to check the return value/exception of `get_jobs_page` and render this into the `index.html` page appropriately (or return a completely different template)

We don't want to start the rendering process just to get the `UndefinedError` from `jinja`. Depending on the `get_jobs_page` call we already know that it's going to fail, thus we should never even call `redner_template`. Do you agree with that?